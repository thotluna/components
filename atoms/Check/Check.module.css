.container-check {
  position: relative;
}

.input-check {
  width: 38px;
  height: 38px;
  appearance: none;
  background: var(--color-primary-highlight);
  border-radius: 20px;
}

.size-sm {
  width: 28px;
  height: 28px;
}

.size-md {
  width: 30px;
  height: 30px;
}

.size-lg {
  width: 38px;
  height: 38px;
}

.color-primary {
  border: 1px solid var(--color-primary);
}

.color-secondary {
  border: 1px solid var(--color-secondary);
}

.icon-check {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
