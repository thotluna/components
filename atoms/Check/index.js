export { default } from './Check'
export { options } from './constants'
export { default as styles } from './Check.module.css'
