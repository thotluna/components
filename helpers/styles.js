import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
