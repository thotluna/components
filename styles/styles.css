@import './minireset.min.css';
@import './tokens.css';
@import './global.css';
@import './variables.css';
