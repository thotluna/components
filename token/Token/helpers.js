import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
