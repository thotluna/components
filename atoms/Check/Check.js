import React from 'react'
import PropTypes from 'prop-types'
import styles from './Check.module.css'
import { options } from './constants'
import Icon from '../../atoms/Icons'
import { getClasses } from '../../helpers/styles'

export const Check = ({ id, isChecked, onChange, name, color, size }) => {
  const getStyles = getClasses(styles)({ size, color })
  return (
    <div className={getStyles('container-check', ['size'])}>
      <input
        id={id}
        className={getStyles('input-check', ['color', 'size'])}
        type="checkbox"
        checked={isChecked}
        aria-label={name}
        tabIndex={0}
        onChange={onChange}
        onKeyPress={onChange}
      />

      {isChecked && (
        <div className={getStyles('icon-check')}>
          <Icon color={color} type="check" size={size} />
        </div>
      )}
    </div>
  )
}

Check.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
  onChange: PropTypes.func,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
}

Check.defaultProps = {
  defaultChecked: false,
  color: 'primary',
  size: 'sm',
}

export default Check
