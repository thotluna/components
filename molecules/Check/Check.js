import React from 'react'
import PropTypes from 'prop-types'
import styles from './Check.module.css'
import withStyles from '../../hocs/withStyles'

import Buttons from '../Buttons'

export const Check = ({ isChecked, getStyles }) => {
  return (
    <Buttons
      ariaLabel="check"
      className={getStyles()}
      isOnlyIcon
      color="inverter"
      size="mini"
      icon={isChecked ? 'check' : null}
    />
  )
}

Check.propTypes = {
  isChecked: PropTypes.bool,
  getStyles: PropTypes.func.isRequired,
}

Check.defaultProps = {
  isChecked: false,
}

export default withStyles(styles)(Check)
