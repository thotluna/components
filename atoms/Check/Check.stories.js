import { Check } from './Check'
import styles from './Check.module.css'
import { options } from './constants'
import {
  getListTemplate,
  getOptionsArgTypes,
  getTemplate,
} from '../../helpers/storybook'

const Template = getTemplate(Check, styles)
const ListTemplate = getListTemplate(Check, styles)

const args = {
  id: 'check#',
  name: 'check',
  defaultChecked: false,
}

export default {
  title: 'Atoms/Check',
  component: Check,
  args,
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
    isChecked: {
      control: {
        type: 'boolean',
      },
    },
  },
}

export const Default = Template.bind({})
Default.args = { ...args, id: 'check:false' }

export const ActiveColor = ListTemplate.bind({})
ActiveColor.args = {
  items: options.colors.map((color) => ({
    ...args,
    id: `true:${color}`,
    defaultChecked: true,
    color,
  })),
}

export const InactiveColor = ListTemplate.bind({})
InactiveColor.args = {
  items: options.colors.map((color) => ({
    ...args,
    id: `false:${color}`,
    color,
  })),
}
