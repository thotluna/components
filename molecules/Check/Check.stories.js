import { Check } from './Check'
import styles from './Check.module.css'
import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(Check, styles)

export default {
  title: 'Molecules/Check',
  component: Check,
  args: {
    isChecked: false,
  },
  argTypes: {
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})
Default.args = { isChecked: false }

export const Active = Template.bind({})
Active.args = { isChecked: true }
