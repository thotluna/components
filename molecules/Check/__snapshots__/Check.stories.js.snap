// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Check Active 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    aria-label="check"
    className="buttons size-mini color-inverter is-only-icon"
  >
    <div
      className="container color-primary size-sm"
    >
      <svg
        fill="none"
        height="100%"
        stroke="currentColor"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 13l4 4L19 7"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={1}
        />
      </svg>
    </div>
  </button>
</div>
`;

exports[`Storyshots Molecules/Check Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    aria-label="check"
    className="buttons size-mini color-inverter is-only-icon"
  />
</div>
`;
